File: ucf_pt/local2025QualifyingRoundData/solutions/judge/judge.cpp
// UCF Local Contest 2025 (Qualifying) - Judge Meetings
// Solution written by Tyler Marks

#include<bits/stdc++.h>
using namespace std;

int main() {
    // Scan in number of coaches and days
    int n, m; cin >> n >> m;

    // store a frequency array storing how many
    // coaches are out on each day
    vector<int> frq(m+1);

    // Scan in each coaches vacations
    for(int i = 0; i < n; i++) {
        int v; cin >> v;
        for(int j = 0; j < v; j++) {
            int s, e; cin >> s >> e;

            // Update frequency for each day in range
            for(int k = s; k <= e; k++)
                frq[k]++;
        }
    }

    // Count how many days have enough coaches
    int res = 0;
    for(int i = 1; i <= m; i++)
        if(n - frq[i] >= 3)
            res++;

    // Output the answer
    cout << res << '\n';
}

File: ucf_pt/local2025QualifyingRoundData/solutions/judge/judge.py
# UCF Local Contest 2025 (Qualifying) - Judge Meeting
# Solution written by Tyler Marks

# scan in the number of coaches and days
n, m = map(int, input().split())

# store a frequency array of how many coaches
# are missing each day
frq = [0] * (m+1)

# scan in each coaches vacation time
for i in range(n):
    ranges = list(map(int, input().split()))
    v = ranges[0]
    for j in range(v):
        s = ranges[2*j+1]
        e = ranges[2*j+2]

        # update frq for each day in range
        for k in range(s, e+1):
            frq[k] += 1

# count how many days have enough coaches
res = 0
for i in range(1, m+1):
    if n - frq[i] >= 3:
        res += 1

# output the answer
print(res)